src/app/course-manager/date-picker/date-picker.component.ts

```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+
+@Component({
+  selector: 'app-date-picker',
+  templateUrl: './date-picker.component.html',
+  styleUrls: ['./date-picker.component.sass']
+})
+export class DatePickerComponent implements OnInit {
+  @Input() parentForm: FormGroup;
+
+  ngOnInit() {
+    console.log(this.parentForm);
+  }
```

Comment:
This component looks like a presentation component. Then you can add `OnPush` strategy to it.